lib/jsdom/living/fetch/Headers-impl.js

```diff
@@ -0,0 +1,231 @@
+"use strict";
+
+const MIMEType = require("whatwg-mimetype");
+
+const {
+  FORBIDDEN,
+  FORBIDDEN_RESPONSE,
+  NO_CORS_SAFELISTED_REQUEST,
+  PRIVILEGED_NO_CORS_REQUEST
+} = require("./reservedHeaders");
+
+const CORS_UNSAFE_BYTE = /[\x00-\x08\x0A-\x1F"():<>?@[\\\]{}\x7F]/;
+
+function assertName(name) {
+  if (!name.match(/^[!#$%&'*+\-.^`|~\w]+$/)) {
+    throw new TypeError("name is invalid");
+  }
+}
+
+function assertValue(value) {
+  if (value.match(/[\0\r\n]/)) {
+    throw new TypeError("value is invalid");
+  }
+}
+
+function isForbiddenHeader(name) {
```

Comment:
It seems like we have two ways of testing a header: certain Sets, and certain functions. Maybe it would be better if the reserved-headers/header-types/whatever file just exported functions, leaving the sets as an encapsulated implementation detail? Then this file could focus mostly on the Headers class itself, not on header types or on header list stuff.